lib/Listeners/LoadPublicViewerListener.php

```diff
@@ -27,22 +27,55 @@
 namespace OCA\Files_PDFViewer\Listeners;
 
 use OCA\Files_PDFViewer\AppInfo\Application;
-use OCP\AppFramework\Http\Events\BeforeTemplateRenderedEvent;
+use OCA\Files_Sharing\Event\BeforeTemplateRenderedEvent;
 use OCP\AppFramework\Http\TemplateResponse;
+use OCP\AppFramework\QueryException;
 use OCP\EventDispatcher\Event;
 use OCP\EventDispatcher\IEventListener;
+use OCP\IServerContainer;
 use OCP\Util;
 
 class LoadPublicViewerListener implements IEventListener {
+
+	/** @var IServerContainer */
+	private $serverContainer;
+
+	public function __construct(IServerContainer $serverContainer) {
+		$this->serverContainer = $serverContainer;
+	}
+
 	public function handle(Event $event): void {
 		if (!$event instanceof BeforeTemplateRenderedEvent) {
 			return;
 		}
 
-		// Make sure we are on a public page rendering
-		if ($event->getResponse()->getRenderAs() !== TemplateResponse::RENDER_AS_PUBLIC) {
+		// If the event has a scope it is not the default share page but, for
+		// example, the authentication page
+		if ($event->getScope() !== null) {
+			return;
+		}
+
+		// Do not load the viewer if there is a download limit
+		if ($this->getDownloadLimit($event->getShare()->getToken()) >= 0) {
 			return;
 		}
+
 		Util::addScript(Application::APP_ID, 'files_pdfviewer-public');
 	}
+
+	private function getDownloadLimit(string $shareToken): int {
+		try {
+			$limitMapper = $this->serverContainer->get('\OCA\Files_DownloadLimit\Db\LimitMapper');
+		} catch (QueryException $e) {
+			return -1;
+		}
+
+		try {
+			$shareLimit = $limitMapper->get($shareToken);
+		} catch (\Exception $e) {
+			return -1;
+		}
+
+		return $shareLimit->getLimit();
+	}
```

Comment:
> I don't like this, this is an anti pattern. Using another app as a dependency is a no-go from me.
> If there is a limit, I would make the Files_DownloadLimit handle the event and do its changes though there ?

Sorry, I did not understand that. How would you prevent the PDF viewer from being loaded by handling the event in the files_downloadlimit app? And wouldn't that make the files_Ddwnloadlimit app also depend on another app?

> How do we manage for images/video for example?

Apparently we do not :shrug: I have just checked and the same issue happens with images and videos. In fact, with videos is even worse, because they are downloaded in several chunks, so just opening the public share page of a video reduces the download limit several times.

Reply:
> However, please note that the reason to implement the check directly in the PDF viewer without touching anything else was to be able to easily backport the fix to previous versions.

@skjnldsv since you objected to cross-app dependency, are you ok with the approach for backportability or have suggestions for further tweaks ?

@danxuliu I think we can go "clean approach" for Nextcloud 26 with the new event on master and keep your code here for Nextcloud <= 25